test/framework/deployments/tracing/universal.go

```diff
@@ -64,12 +64,12 @@ func getIP(t testing.TestingT, container string) (string, error) {
 }
 
 func (j *universalDeployment) allocatePublicPortsFor(ports ...string) error {
-	for i, port := range ports {
-		pubPortUInt32, err := util_net.PickTCPPort("", uint32(33204+i), uint32(34204+i))
+	for _, port := range ports {
+		pubPort, err := utils.GetFreePort()
```

Comment:
there's probably some logic in the tests that prevents this but can we not let docker pick a random free port on the host? Using `--publish=<container-port>` and then use `docker port` or `inspect` to retrieve the chosen ports?

Reply:
that's a good idea, let me try that